src/Bicep.Core/TypeSystem/DeployTimeConstantVisitor.cs

```diff
@@ -91,13 +91,14 @@ public override void VisitObjectSyntax(ObjectSyntax syntax)
                 return;
             }
             // Only visit the object properties if they are required to be deploy time constant.
-            foreach (var deployTimeIdentifier in ObjectSyntaxExtensions.ToNamedPropertyDictionary(syntax))
+            foreach (var propertyName in ObjectSyntaxExtensions.ToKnownPropertyNames(syntax))
```

Comment:
I'm not 100% sure right now why this change was needed .... I think at some point I had a case where this was throwing due to duplicate names, in an existing test.

Anyways this code was relatively fragile, because given error cases it would trigger an exception.

Reply:
I suspect this might fix #1432 👍